src/main/java/com/company/KafkaProducerTest.java

```diff
@@ -41,17 +41,17 @@ public static void main(String[] args) {
     props.put("key.serializer", "org.apache.kafka.common.serialization.StringSerializer");
     props.put("value.serializer", "org.apache.kafka.common.serialization.StringSerializer");
     Producer<String, String> producer = new KafkaProducer(props);
-
-//    for (int i = 0; i < 100; i++) {
-//      producer.send(new ProducerRecord<String, String>("my-topic", Integer.toString(i), Integer.toString(i)));
-//    }
-    LogReader(producer);
+    logReader(producer);
     producer.close();
   }
+  /** new function
+   * reads the apache access log file, line by line
+   * send each line as a message (before to the activeMQ broker)
+   * close the file and exit the producer thread
+   */
+  public static void logReader(Producer<String, String>  producer) throws IOException {
 
-  public static void LogReader(Producer<String, String>  producer) {
-
-  try
+      try
```

Comment:
I might be totally wrong about the changes, but I am trying to make this work. I appreciate if you let me know my mistakes